client/src/components/TypeList.vue

```diff
@@ -1,11 +1,13 @@
 <script lang="ts">
-import { defineComponent, reactive } from '@vue/composition-api';
+import { computed, defineComponent, reactive } from '@vue/composition-api';
 import { useCheckedTypes, useAllTypes, useTypeStyling } from '../provides';
 
 export default defineComponent({
   name: 'TypeList',
 
-  setup(props, { emit }) {
+  setup(props, { emit, root }) {
+    const prompt = root.$prompt;
```

Comment:
Can't use the prompt here.    Same as above: you could replace the button with a slot of a self-contained delete button using

```
const checkedTypesRef = useCheckedTypes();
const allTypesRef = useAllTypes();
```

The upshot is then you can stick this thing anywhere.  That's the nice those `use()` funcs let you do.

So you'll just have a `DeleteSelectedTracks.vue` and a `DeleteSelectedTypes.vue` and they both just render a button and you slot them into these components, but there's no shared state between them.